### ISSUE TITLE
Used as an alias, `ClassVar` is not correctly identified

### ISSUE THREAD

**Issue by Diogo-Rossi** [microsoft/pyright#7401]
The following snippet works fine, no issue raised:
```python
from dataclasses import dataclass
from typing import ClassVar


@dataclass
class Test:
    a: str
    b: int
    c: ClassVar[str]


t = Test("word", 33)

```
But, when using `ClassVar` in an alias, it raises a wrong issue
```python
from dataclasses import dataclass
from typing import ClassVar

Alias = ClassVar[str]


@dataclass
class Test:
    a: str
    b: int
    c: Alias


t = Test("word", 33) # ISSUE: Argument missing for parameter "c"

```
The 2 snippets above works ok at runtime.
But if you try to solve the issue in the second snippet, an expected error happens, because `c` is a `ClassVar`

```python
...
t = Test("word", 33, 'test') # no issue anymore with pyright

# output at runtime:
#    t = Test("word", 33, 'test')
#        ^^^^^^^^^^^^^^^^^^^^^^^^
# TypeError: Test.__init__() takes 3 positional arguments but 4 were given
```

**VS Code extension or command-line**
VS Code extension integrated into Pylance
VSCode version: 1.87.0
Pylance Version: 2024.2.3

**Comment by erictraut** [comment]
Pyright's behavior here is correct. `ClassVar` is not a valid type form and cannot be used in a type alias. It is a special form that can be used only within a class body.

**Comment by Diogo-Rossi** [comment]
> Pyright's behavior here is correct. `ClassVar` is not a valid type form and cannot be used in a type alias. It is a special form that can be used only within a class body.

I understand. Thanks for the explanation.
For reference: also see discussions in: [1172#discussion-4045503](https://github.com/python/typing/discussions/1172#discussion-4045503) and #3412 .

---

But I should make a observation:
This behaviour would make it unavailable to change symbol names inside python code in normal way.

For instance, take both snippets of my original post: they behave the same in run time.

Also, this following example works fine and it is correctly interpreted by pylance/pyright:

```python
from typing import ClassVar as Alias
from dataclasses import dataclass

@dataclass
class Test:
    a: str
    b: Alias
    c: int

t = Test('test', 32) # no issue here, as "b" is a "ClassVar"
```
Also see related: #7357
